launcher/cog-launcher.c

```diff
@@ -1271,11 +1336,22 @@ cog_launcher_handle_local_options(GApplication *application, GVariantDict *optio
 
 #if HAVE_WEBKIT_NETWORK_PROXY_API
     if (s_options.proxy) {
-        WebKitWebsiteDataManager             *data_manager = cog_launcher_get_web_data_manager(launcher);
+        if (launcher->automated) {
+            g_printerr("%s: Cannot set proxy settings in automation mode\n", g_get_prgname());
+            return EXIT_FAILURE;
+        }
+
         g_autoptr(WebKitNetworkProxySettings) webkit_proxy_settings =
             webkit_network_proxy_settings_new(s_options.proxy, (const gchar *const *) s_options.ignore_hosts);
-        webkit_website_data_manager_set_network_proxy_settings(data_manager, WEBKIT_NETWORK_PROXY_MODE_CUSTOM,
+#if COG_USE_WPE2
+        webkit_network_session_set_proxy_settings(launcher->network_session,
+                                                  WEBKIT_NETWORK_PROXY_MODE_CUSTOM,
+                                                  webkit_proxy_settings);
```

Comment:
Having some things like the proxy settings in something called “network session” does indeed make sense and the new API feels nicer in general 👍🏼